### PR TITLE
htmlq: init at 0.2.0

### LINKED ISSUE #137037: htmlq
**Project description**

Like jq, but for HTML. Uses CSS selectors to extract bits of content from HTML files.

**Metadata**

* homepage URL: https://github.com/mgdm/htmlq
* source URL: https://github.com/mgdm/htmlq
* license:  MIT
* platforms: Unix

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Closes #137037.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 137064` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>htmlq</li>
  </ul>
</details>

**Comment by NerdyPepper** [comment]
Oh looks like we ended up working on the same thing. Since you have already tested on darwin, I'd say we keep this over #137067. 

Would you mind adding `doCheck = false` since `htmlq` does not have tests at the moment?

**Comment by siraben** [comment]
Done.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5973,6 +5973,10 @@ with pkgs;
 
   html-proofer = callPackage ../tools/misc/html-proofer { };
 
+  htmlq = callPackage ../development/tools/htmlq {
+    inherit (pkgs.darwin.apple_sdk.frameworks) Security;
```
Comment (NerdyPepper) [inline pkgs/top-level/all-packages.nix:5977]:
```suggestion
    inherit (darwin.apple_sdk.frameworks) Security;
```

I think `pkgs` is not required.

**Comment by NerdyPepper** [comment]
Looks good to me!

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (siraben) [reply pkgs/top-level/all-packages.nix:5977]: Oh I just saw this comment. Looks like there are several occurrences of this in `all-packages.nix` so a file-wide change might be in order.

Comment (NerdyPepper) [reply pkgs/top-level/all-packages.nix:5977]: Ah, doesn't look like there is much of a convention.  

>Looks like there are several occurrences of this in all-packages.nix so a file-wide change might be in order.

Good call.

**Comment by siraben** [comment]
@NerdyPepper feel free to add yourself to the maintainers list.

**Comment by NerdyPepper** [comment]
I will take you up on that offer :D

**Pull request merged**